### PR TITLE
Refactor update circuit version

### PR DESCRIPTION
**Pull request by FoodChain1028**
## Summary

Refactor and update the circuit to `circom 2.1.0`. 

## Linked Issue

#70

## Details

There are 2 main part changed:
1. `packages/circuits/circuits/dataProof.circom` refactoring and the files built from snark.
2. `packages/circuits/package.json` adding a script (`buildOverwrite`) to rebuilt the snark files based on the new circuit. 

## Impacted Areas

The main changed file would be `packages/circuits/circuits/dataProof.circom`, with the circuit based snark file changed. (you can find the built files in `packages/circuits/zkBuild` directory) 

## Tests

The test is generated from `create-unirep-app`.
- Prove data in Unirep App
    - should generate a data proof
    - should generate a data proof with values
    - should not generate a data proof with invalid values 

## Visual Materials

no

## Verification Steps

First, at the root dir, run
```
yarn install && yarn build
```

Which would build the files you needed for test.

After that, run
```
yarn circuits test
```
to run the tests of circuits.

The result should pass 3 tests:

```
yarn circuits test                                                                                                                  ─╯
yarn run v1.22.19
$ yarn workspace @unirep-app/circuits run test
$ mocha -r ts-node/register test/*.test.ts --exit


  Prove data in Unirep App
Gen Proof time: 1081 ms (1 s)
    ✔ should generate a data proof (1444ms)
Gen Proof time: 962 ms (0 s)
    ✔ should generate a data proof with values (1335ms)
ERROR:  4 Error in template DataProof_78 line: 45

    ✔ should not generate a data proof with invalid values (66ms)


  3 passing (3s)

✨  Done in 5.31s.
```

## Todo

Break down the progress of the PR for everyone to see what else you intend to include in this PR.

-   [X] refactor the circuit 
-   [X] added a script to rebuild the snark-related files

## Checklist

-   [X] All new and existing tests pass
-   [X] I have updated the documentation (if applicable)
-   [X] My changes do not introduce new security vulnerabilities

### CODE REVIEW

File: packages/circuits/circuits/dataProof.circom [hunk 1]
```diff
@@ -1,9 +1,7 @@
-pragma circom 2.0.0;
+pragma circom 2.1.0;
 
-include "../../../node_modules/@unirep/circuits/circuits/proveReputation.circom";
-include "../../../node_modules/@unirep/circuits/circuits/circomlib/circuits/poseidon.circom";
-include "../../../node_modules/@unirep/circuits/circuits/circomlib/circuits/mux1.circom";
-include "../../../node_modules/@unirep/circuits/circuits/circomlib/circuits/gates.circom";
+include "../../../node_modules/@unirep/circuits/circuits/hasher.circom";
```
Comment (ryanycw) [inline packages/circuits/circuits/dataProof.circom:3]: And I believe "../../../node_modules/" could be removed to make the style cleaner.

**Review by ryanycw — APPROVED** [review]
Thank you for the great work! LGTM, only some minor issues.

**Review by ryanycw — APPROVED** [review]
Thank you for the great work! LGTM, only some minor issues.

File: packages/circuits/package.json [hunk 2]
```diff
@@ -13,6 +13,7 @@
     },
     "scripts": {
         "build": "tsc && yarn copyCircom",
+        "buildOverwrite": "find ./zksnarkBuild -type f -not -name '*.ptau' -delete && yarn buildsnark",
```
[the remark this answers is not in the document]
Comment (FoodChain1028) [reply packages/circuits/package.json:16]: Yes, the reason of this is due to the phase2 setup with circuit (if you already build with the origin circuit and modified the circuit later, this command would work to rebuild the files inside `zkSnarkBuild` ), I will add this to `README.md` later

**Comment by FoodChain1028** [comment]
> Thank you for the great work! LGTM, only some minor issues.

Thank you for your review. The issues you mentioned have been modified and tested.

**Review by ryanycw — APPROVED** [review]
LGTM!

**Pull request merged**